File: hospital-appointment-frontend/src/api/hospital/patient.ts
/**
 * 患者管理 API
 */

import request from '../request'

export interface Patient {
  patientID: number
  patientName: string
  idCard: string
  patientGender: number
  patientBirthday?: string
  patientPhone: string
  patientEmail?: string
  patientAddress?: string
  medicalCardNo?: string
  insuranceType?: number
  insuranceNo?: string
  allergyHistory?: string
  medicalHistory?: string
  emergencyContact?: string
  emergencyPhone?: string
  creditScore: number
  noshowCount: number
  isBlacklist: number
  patientStatus: number
}

export const patientAPI = {
  /**
   * 患者注册
   */
  register: (data: Partial<Patient>) => {
    return request.post<string>('/hospital/patient/register', data)
  },

  /**
   * 患者登录
   */
  login: (phone: string, password: string) => {
    return request.post<Patient>('/hospital/patient/login', null, { params: { phone, password } })
  },

  /**
   * 获取患者信息
   */
  getById: (id: number) => {
    return request.get<Patient>(`/hospital/patient/${id}`)
  },

  /**
   * 更新患者信息
   */
  update: (id: number, data: Partial<Patient>) => {
    return request.put<string>(`/hospital/patient/${id}`, data)
  },

  /**
   * 修改密码
   */
  changePassword: (id: number, oldPassword: string, newPassword: string) => {
    return request.put<string>(`/hospital/patient/${id}/password`, null, {
      params: { oldPassword, newPassword }
    })
  },

  /**
   * 检查是否在黑名单
   */
  isInBlacklist: (id: number) => {
    return request.get<boolean>(`/hospital/patient/${id}/blacklist`)
  },

  /**
   * 获取患者信用积分
   */
  getCreditScore: (id: number) => {
    return request.get<number>(`/hospital/patient/${id}/credit`)
  }
}

File: hospital-appointment-backend/src/main/java/com/hospital/appointment/entity/Patient.java
package com.hospital.appointment.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 患者实体类
 * 改造自学生用户(studentuser)
 */
@Data
@TableName("patient")
public class Patient implements Serializable {
    
    @TableId(type = IdType.AUTO)
    private Long patientID;
    
    /** 患者姓名 */
    private String patientName;
    
    /** 登录密码 */
    private String patientPassword;
    
    /** 身份证号 */
    private String idCard;
    
    /** 性别：1-男 2-女 */
    private Integer patientGender;
    
    /** 出生日期 */
    private LocalDate patientBirthday;
    
    /** 手机号 */
    private String patientPhone;
    
    /** 邮箱 */
    private String patientEmail;
    
    /** 住址 */
    private String patientAddress;
    
    /** 就诊卡号 */
    private String medicalCardNo;
    
    /** 医保类型：1-城镇职工 2-城乡居民 3-自费 */
    private Integer insuranceType;
    
    /** 医保卡号 */
    private String insuranceNo;
    
    /** 过敏史 */
    private String allergyHistory;
    
    /** 既往病史 */
    private String medicalHistory;
    
    /** 紧急联系人 */
    private String emergencyContact;
    
    /** 紧急联系电话 */
    private String emergencyPhone;
    
    /** 信用积分 */
    private Integer creditScore;
    
    /** 爽约次数 */
    private Integer noshowCount;
    
    /** 是否黑名单：0-否 1-是 */
    private Integer isBlacklist;
    
    /** 账号状态：0-禁用 1-正常 */
    private Integer patientStatus;
    
    /** 注册时间 */
    private LocalDateTime createTime;
}
